### PR TITLE
Upgrade dependencies for 1.31.0

### PR DESCRIPTION
**Pull request by jrhee17**
### Dependencies

- Brotli4j 1.16.0 → 1.17.0
- java-control-plane 1.0.45 → 1.0.46
- Eureka 2.0.3 → 2.0.4
- fastutil 8.5.14 → 8.5.15
- GraphQL Kotlin 7.1.4 → 8.2.1
- Java gRPC 1.65.1 → 1.68.1
- Guava 33.2.1-jre → 33.3.1-jre
- Jackson 2.17.2 → 2.18.1
- Jetty
  - 11.0.22 → 11.0.24
  - 12.0.12 → 12.0.14
- Kotlin 3.8.0 → 3.8.1
- Kotlin coroutines 1.8.1 → 1.9.0
- Fabric8 Kubernetes Client 6.13.1 → 6.13.4
- Micrometer 1.13.2 → 1.13.6
- Micrometer Tracing 1.3.2 → 1.3.5
- Netty 4.1.112.Final → 4.1.114.Final
- prometheus 1.3.1 → 1.3.2
- Protobuf 3.25.1 → 3.25.5
- protobuf-jackson 2.5.0 → 2.6.0
- Reactor 3.6.8 → 3.6.11
- RXJava 3.1.8 → 3.1.9
- Sangria 4.1.1 → 4.2.2
- Scala
  - 2.12.19 → 2.12.20
  - 2.13.14 → 2.13.15
  - 3.4.2 → 3.6.1
- Spring 6.1.11 → 6.1.14
- Spring Boot 3.3.2 → 3.3.5
- Tomcat
  - 9.0.91 → 9.0.96
  - 10.1.26 → 10.1.31

### Dev dependencies
- apache-httpclient5 5.3.1 → 5.4.1
- asm 9.7 → 9.7.1
- awaitility 4.2.1 → 4.2.2
- blockhound 1.0.9.RELEASE → 1.0.10.RELEASE
- curator 5.7.0 → 5.7.1
- dagger 2.51.1 → 2.52
- dropwizard-metrics 4.2.26 → 4.2.28
- errorprone 2.29.2 → 2.35.1
- errorprone-gradle-plugin 4.0.1 → 4.1.0
- eureka 2.0.3 → 2.0.4
- fastutil 8.5.14 → 8.5.15
- gax-grpc 2.51.0 → 2.57.0
- hamcrest 2.2 → 3.0
- jkube 1.16.2 → 1.17.0
- junit5 5.10.3 → 5.11.3
- kafka 3.8.0 → 3.8.1
- munit 1.0.0 → 1.0.2
- node-gradle-plugin 7.0.2 → 7.1.0
- nullaway 0.11.1 → 0.12.1
- nullaway-gradle-plugin 2.0.0 → 2.0.1
- snappy 1.1.10.5 → 1.1.10.7
- slf4j2 2.0.13 → 2.0.16
- testcontainers 1.20.1 → 1.20.3

### Additional changes
- Added an integration test module `logback15` which runs tests in the `logback14` module with `logback15` to ensure compatibility.
- `DropwizardExtensionsSupport` isn't compatible with recent JUnit versions following the removal of `ReflectionUtils#makeAccessible(AccessibleObject)`. The `before` method is called manually.
- Minor breaking changes in the test classes for kubernetes client
- Scala upgrade enforces the usage of the `using` keyword
- Tomcat's `AbstractProcessor` has added a `earlyHints` method which needs to be implemented
- Miscellaneous refactoring due to errorprone upgrade

### CODE REVIEW

**Review by minwoox — APPROVED** [review]
👍 👍 👍

File: scalapb/scalapb_3/src/main/scala/com/linecorp/armeria/server/scalapb/ScalaPbRequestConverterFunction.scala [hunk 1]
```diff
@@ -173,7 +173,7 @@ final class ScalaPbRequestConverterFunction private (jsonParser: Parser, resultT
    */
   private def jsonToScalaPbMessage(expectedResultType: Class[_], json: String): Any with Serializable = {
     val messageType = extractGeneratedMessageType(expectedResultType)
-    val message: GeneratedMessage = jsonParser.fromJsonString(json)(getCompanion(messageType))
+    val message: GeneratedMessage = jsonParser.fromJsonString(json)(using getCompanion(messageType))
```
Comment (ikhoon) [inline scalapb/scalapb_3/src/main/scala/com/linecorp/armeria/server/scalapb/ScalaPbRequestConverterFunction.scala:176]: 👍

**Review by ikhoon — APPROVED** [review]
Thanks! 👍	

Optional)Would you also upgrade Gradle dependency? That could be a preliminary step before upgrading Java version to 23.

**Review by trustin — COMMENTED** [review]
:+1::+1::+1:

**Pull request merged**